src/PeerReview/SubmissionRepo/FPCourse.hs

```diff
@@ -1,24 +1,59 @@
 {-# LANGUAGE OverloadedStrings #-}
 module PeerReview.SubmissionRepo.FPCourse
     ( repo
+    , repoTest
     ) where
 
 import           PeerReview.Types
+import           Data.Aeson
+import           Data.Vector                     as V
+import           Data.Map                        as M
+import           Data.Text
+import           Control.Monad
+import qualified PeerReview.SubmissionRepo.FPCourseAPIClient as FPCourseAPIClient
 
 -- Create submission repo using endpoint configuration.
 -- Config is a map containing required endpoints for fetching data.
 repo :: SubmissionRepoConfig -> SubmissionRepo
-repo _ = SubmissionRepo byId forTask forUser listAll
+repo cfg =
+    let client = FPCourseAPIClient.client
+    in SubmissionRepo (byId cfg client) (forTask cfg client) (forUser cfg client) (listAll cfg client)
 
+-- For testing purposes
+repoTest :: APIClient -> SubmissionRepoConfig -> SubmissionRepo
+repoTest client cfg =
+    SubmissionRepo (byId cfg client) (forTask cfg client) (forUser cfg client) (listAll cfg client)
 
-byId :: SubmissionID -> IO Submission
-byId _ = return $ Submission "1" "2" "3" (Just "wat")
+byId :: SubmissionRepoConfig -> APIClient ->SubmissionID -> IO Submission
+byId _ _ _ = return $ Submission "1" "2" "3" (Just "wat")
 
-forTask :: TaskID -> IO [Submission]
-forTask _ = return []
+forTask :: SubmissionRepoConfig -> APIClient -> TaskID -> IO [Submission]
+forTask cfg client taskId = do
+    allSubmissions <- getAllSubmissions cfg client
+    let subsForTask = liftM (V.filter (\s -> sTid s == taskId)) allSubmissions
+    case subsForTask of
+        Just submissions -> return $ V.toList submissions
+        _                -> return []
 
-forUser :: UserID -> IO [Submission]
-forUser _ = return []
+forUser :: SubmissionRepoConfig -> APIClient -> UserID -> IO [Submission]
+forUser _ _ _ = return []
 
-listAll :: IO [Submission]
-listAll = return []
+listAll :: SubmissionRepoConfig -> APIClient -> IO [Submission]
+listAll cfg client = do
+    allSubmissions <- getAllSubmissions cfg client
+    case allSubmissions of
```

Comment:
`maybe` could be used here instead of `case` https://hackage.haskell.org/package/base-4.8.1.0/docs/Prelude.html#v:maybe